lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -33,6 +34,19 @@ public static ParsedObject extractKeywords(
             final ObjectLookupProvider lookup,
             final List<GUID> objectRefPath)
             throws IOException, ObjectParseException, IndexingException, InterruptedException {
+
+        // check pre-conditons
+        if(json == null) throw new IllegalArgumentException("json is a required parameter");
```

Comment:
I'm curious to know if it's really necessary to pass around json strings rather than objects. Does the code keep a ton of strings in memory at once? I'm assuming the strings aren't that big...?

Reply:
I don't think there are that many json strings being held in memory and they can be represented as objects. I have made a note to look closer at this and do the refactoring for this a little down the line.